sys/log/full/src/log.c

```diff
@@ -1005,3 +1070,21 @@ log_set_max_entry_len(struct log *log, uint16_t max_entry_len)
     assert(log);
     log->l_max_entry_len = max_entry_len;
 }
+
+int
+log_fill_current_img_hash(struct log_entry_hdr *hdr)
+{
+#if MYNEWT_VAL(LOG_FLAGS_IMAGE_HASH)
+    int rc;
+
+    hdr->ue_flags |= LOG_FLAGS_IMG_HASH;
+
+    /* We have to account for LOG_IMG_HASHLEN bytes of hash */
+    rc = imgr_get_current_hash(hdr->ue_imghash, LOG_IMG_HASHLEN);
+    if (rc) {
+        return rc;
+    }
+
+#endif
+    return SYS_ENOTSUP;
```

Comment:
If `LOG_FLAGS_IMAGE_HASH` is not enabled, I think this function should populate the hash field with a known value (all zeros? all 0xff?).  Otherwise, if a device writes some logs with the hash, then gets upgraded to an image that does not support this feature, a log read will yield entries with indeterminate hashes.